From 97343e0e5cbcdccca7c1f3b2837ceca5f94c9abb Mon Sep 17 00:00:00 2001
From: Naveen Mamindlapalli <naveenm@marvell.com>
Date: Wed, 18 Nov 2020 20:29:19 +0530
Subject: [PATCH 0937/1921] octeontx2-bphy-netdev: Fix list_entry usage in the
 code

Fix list_entry usage to get the ptp_skb_list node element
structure instead of head of the list.

Change-Id: I9a3f49429294581100b6d31d1103706f081086c1
Signed-off-by: Naveen Mamindlapalli <naveenm@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/40355
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c
index bc8cc278db34..4ddf6dc09a39 100644
--- a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c
+++ b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c
@@ -226,6 +226,7 @@ static void otx2_rfoe_ptp_submit_work(struct work_struct *work)
 	struct ptp_tstamp_skb *ts_skb;
 	u16 psm_queue_id, queue_space;
 	struct sk_buff *skb = NULL;
+	struct list_head *head;
 	u64 jd_cfg_ptr_iova;
 	unsigned long flags;
 	u64 regval;
@@ -260,8 +261,8 @@ static void otx2_rfoe_ptp_submit_work(struct work_struct *work)
 		return;
 	}
 
-	ts_skb = list_entry(&priv->ptp_skb_list.list, struct ptp_tstamp_skb,
-			    list);
+	head = &priv->ptp_skb_list.list;
+	ts_skb = list_entry(head->next, struct ptp_tstamp_skb, list);
 	skb = ts_skb->skb;
 	list_del(&ts_skb->list);
 	kfree(ts_skb);
-- 
2.31.1

